Exponential growth and decay model, and more function operations.

steverod opened the pull request:
At long last, the theorems discussed at <https://groups.google.com/forum/#!topic/metamath/CKF4bpLaH0I>, and a few function operations.

Also ` ssrecnpr ` and (as discussed at <https://groups.google.com/forum/#!topic/metamath/cjxfDdy4yyc>) ` ssdifeq0 `, but since they nor the ` ofmul0or ` theorem I had in mind were not needed for ` expgrowth ` after all (and I'm not sure ` ofmul0or ` as I had thought of it was even *provable*)  I probably won't work with those two further.

The reference [Strang] in expgrowth is:

* Strang, Gilbert, Calculus, 1st ed., Wellesley-Cambridge Press (1991) [QA???]; ISBN 978-09802327-45, available at http://ocw.mit.edu/resources/res-18-001-calculus-online-textbook-spring-2005/.

~~The (in my opinion, fittingly) long description of expgrowth uses the ` y' ` symbol from Jonathan Ben-Naim's mathbox; seemed to make sense there.~~  If there are better sources etc., I certainly don't mind changes to that wall of text.

nmegill commented:
Hi Steve,

For cross referencing book variable names to set.mm names, I have
typically been using black letters for the book's variable names,
especially if they differ from the book's names.  For example, see
theorems arglem1 and php.  I think it is often clearer this way.  (I
should have italicized the book letters in arglem1; I'll do that
eventually.)  If we want to get nitpicky, there is a subtle distinction
between book and Metamath variables, which is that the latter are
metavariables ranging over the former.  In addition, we almost always
use upper-case class variables for free variables.

To italicize a letter such as y, you can use _y_.  The rules are
currently kind of vague for distinguishing underscores for italicizing
and an underscore for subscript (I've been adjusting them as needed but
have no spec yet), but a digit following the underscore will be treated
as a subscript.  So _y__1 will be italicized y with subscript 1 and _y_'
will be italicized y'.  In some cases you just have to try it and see
what happens, and we can discuss it if the behavior is not what you
need.

As for JBN's y', in Strang's case the ' does not identify a new variable
but is a kind of a unary operation on y (treated as a function).  Of
course in this case it is mainly an appearance thing since the y' is not
being used as a true Metamath variable.

BTW I would encourage you to run 'verify markup */f' in the most recent
metamath.exe to check for markup errors.

It it interesting that MIT put Strang's book online, since it seems the
going price for introductory calculus books are in the hundreds of
dollars these days, with a new edition every year to discourage the used
book market.  It is unfortunate the scan was rather poor (I've used the
machine they used; it's in the MIT science library and does awful scans
exactly like that.)  Or maybe they want a poor scan on purpose to
encourage people to buy it.  :)

Norm


On 11/24/15 8:38 PM, steverod wrote:
> At long last, the theorems discussed at
> https://groups.google.com/forum/#!topic/metamath/CKF4bpLaH0I, and a few
> function operations.
>
> Also |ssrecnpr| and (as discussed at
> https://groups.google.com/forum/#!topic/metamath/cjxfDdy4yyc)
> |ssdifeq0|, but since they nor the |ofmul0or| theorem I had in mind were
> not needed for |expgrowth| after all (and I'm not sure |ofmul0or| as I
> had thought of it was even /provable/) I probably won't work with those
> two further.
>
> The reference [Strang] in expgrowth is:
>
>   * Strang, Gilbert, Calculus, 1st ed., Wellesley-Cambridge Press (1991)
>     [QA???]; ISBN 978-09802327-45, available at
>     http://ocw.mit.edu/resources/res-18-001-calculus-online-textbook-spring-2005/.
>
> The (in my opinion, fittingly) long description of expgrowth uses the
> |y'| symbol from Jonathan Ben-Naim's mathbox; seemed to make sense
> there. If there are better sources etc., I certainly don't mind changes
> to that wall of text.

steverod commented:
Well I think I reformatted it more closely to the other two theorems now, and fixed a couple other issues besides, so I think I've addressed that.

The MIT machines, OTOH, are for others to fix. :)
